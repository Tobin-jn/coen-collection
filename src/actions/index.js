export const loadMovies = movies => ({
  type: "LOAD_MOVIES",
  movies
});

export const logIn = loggedIn => ({
  type: "TOGGLE_LOGIN",
  loggedIn: false
});
