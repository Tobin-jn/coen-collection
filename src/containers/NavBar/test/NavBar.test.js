import React from "react";
import { mapStateToProps, mapDispatchToProps, NavBar } from "../index";
import { shallow } from "enzyme";
import { NavLink } from "react-router-dom";
import { logIn, setErrorMessage, resetFavorites, displayFavorites } from "../../../actions";

describe("NavBar", () => {
  let wrapper;
  const mockHandleLogin = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <NavBar
        loggedIn={true}
        currentUser={{ name: "Taylor", id: 2 }}
        handleLogin={mockHandleLogin}
        errorMessage={"Invalid login"}
        showFavorites={false}
        handleLogin={jest.fn()}
        handleErrorMessage={jest.fn()}
        handleResetFavorites={jest.fn()}
        handleDisplayFavorites={jest.fn()}
      />
    );
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render navlinks if user is not logged in", () => {
    wrapper = shallow(
      <NavBar
        loggedIn={false}
        currentUser={{ name: "Taylor", id: 2 }}
        handleLogin={mockHandleLogin}
        errorMessage={"Invalid login"}
        showFavorites={false}
        handleLogin={jest.fn()}
        handleErrorMessage={jest.fn()}
        handleResetFavorites={jest.fn()}
        handleDisplayFavorites={jest.fn()}
      />
    );
    expect(wrapper.find(NavLink).length).toEqual(2);
  });

  it("should render a signout button if user is logged in", () => {
    expect(wrapper.find(NavLink).length).toEqual(2);
    expect(wrapper.find(".sign-out-button")).toBeDefined();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a loggedIn status", () => {
    const mockState = {
      loggedIn: true,
    };

    const expected = {
      loggedIn: true,
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
  it("should return an object with a currentUser", () => {
    const mockState = {
      currentUser: {name: 'Taylor', id: 5},
    };

    const expected = {
      currentUser: {name: 'Taylor', id: 5},
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
  it("should return an object with an errorMessage", () => {
    const mockState = {
      errorMessage: 'Error',
    };

    const expected = {
      errorMessage: 'Error',
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
  it("should return an object with showFavorites status", () => {
    const mockState = {
      showFavorites: true
    };

    const expected = {
      showFavorites: true
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  it("should call dispatch with a logIn action when handleLogin is called", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = logIn(true);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleLogin(true);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
    it("should call dispatch with a setErrorMessage action when handleErrorMessage is called", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setErrorMessage('error');

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleErrorMessage('error');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
      it("should call dispatch with a resetFavorites action when handleResetFavorites is called", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = resetFavorites();

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleResetFavorites();

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
        it("should call dispatch with a displayFavorites action when handleDisplayFavorites is called", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = displayFavorites(true);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleDisplayFavorites(true);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
