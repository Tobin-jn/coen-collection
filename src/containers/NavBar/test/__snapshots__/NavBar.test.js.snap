// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    currentUser="Taylor"
    handleLogin={[MockFunction]}
    loggedIn={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="favorites-button buttons"
          onClick={[Function]}
          to="/favorites"
        >
          My Favorites
        </NavLink>,
        <h1
          className="user-name"
        >
          Hello 
          !
        </h1>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="sign-out-button buttons"
          onClick={[Function]}
          to="/"
        >
          Sign Out
        </NavLink>,
      ],
      "className": "navBar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "My Favorites",
          "className": "favorites-button buttons",
          "onClick": [Function],
          "to": "/favorites",
        },
        "ref": null,
        "rendered": "My Favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            undefined,
            "!",
          ],
          "className": "user-name",
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          undefined,
          "!",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Sign Out",
          "className": "sign-out-button buttons",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="favorites-button buttons"
            onClick={[Function]}
            to="/favorites"
          >
            My Favorites
          </NavLink>,
          <h1
            className="user-name"
          >
            Hello 
            !
          </h1>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="sign-out-button buttons"
            onClick={[Function]}
            to="/"
          >
            Sign Out
          </NavLink>,
        ],
        "className": "navBar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "My Favorites",
            "className": "favorites-button buttons",
            "onClick": [Function],
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "My Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello ",
              undefined,
              "!",
            ],
            "className": "user-name",
          },
          "ref": null,
          "rendered": Array [
            "Hello ",
            undefined,
            "!",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Sign Out",
            "className": "sign-out-button buttons",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
